runtime/util/vmargs.c

```diff
@@ -76,8 +76,41 @@
 
 #define LARGE_STRING_BUF_SIZE 256
 
+#define JAVA_CLASS_PATH_EQUALS "-Djava.class.path="
+#define SUN_JAVA_COMMAND_EQUALS "-Dsun.java.command="
+#define SUN_JAVA_LAUNCHER_PID_EQUALS "-Dsun.java.launcher.pid="
+#define SUN_JAVA_LAUNCHER_EQUALS "-Dsun.java.launcher="
+
+struct JavaLauncherOption {
+	const char *optionName;
+	int optionNameLength;
+};
+
+/* Options in this table will be not treated as user arguments */
+static const struct JavaLauncherOption JavaLauncherOptionTable[] = {
+	{JAVA_CLASS_PATH_EQUALS, LITERAL_STRLEN(JAVA_CLASS_PATH_EQUALS)},
+	{SUN_JAVA_COMMAND_EQUALS, LITERAL_STRLEN(SUN_JAVA_COMMAND_EQUALS)},
+	{SUN_JAVA_LAUNCHER_PID_EQUALS, LITERAL_STRLEN(SUN_JAVA_LAUNCHER_PID_EQUALS)},
+	{SUN_JAVA_LAUNCHER_EQUALS, LITERAL_STRLEN(SUN_JAVA_LAUNCHER_EQUALS)},
+	{VMOPT_XJCL_COLON, LITERAL_STRLEN(VMOPT_XJCL_COLON)}, /* macro from jvminit.h */
+	{NULL, 0} /* The last entry in the table must be Null in order for the search function to exit.*/
+};
+
 static char * getStartOfOptionValue(J9VMInitArgs* j9vm_args, IDATA element, const char *optionName);
 
+/* Binary Search function that searches if optionName is in JavaLauncherOptionTable */
+BOOLEAN
+isGeneratedArgument(const char *optionName) {
```

Comment:
I've decided to switch to a linear search as requested by Keith for the following:
-It might be overkill to implement a binary search since were only reducing the number of loops we’re doing by 2 (MAX)
-With a linear search, the only variable we need to declare is a pointer, which is much cleaner than having a start/end/mid variable.
-The table will not need to be in Lexical Order